### PR TITLE
[FLINK-14901] Throw Error in MemoryUtils if there is problem with using system classes over reflection

### PR DESCRIPTION
**Pull request by azagrebin**
A follow-up hotfix for #9747 and #9997.
If something goes wrong using system classes over reflection n MemoryUtils or JavaGcCleanerWrapper, Flink cannot really recover from it. We should throw an Error in this case which should stay unhandled.

### CODE REVIEW

File: flink-core/src/main/java/org/apache/flink/util/JavaGcCleanerWrapper.java [hunk 1]
```diff
@@ -69,10 +70,12 @@ public static Runnable create(Object owner, Runnable cleanOperation) {
 		return CLEANER_FACTORY.create(owner, cleanOperation);
 	}
 
+	@FunctionalInterface
```
Comment (GJL) [inline flink-core/src/main/java/org/apache/flink/util/JavaGcCleanerWrapper.java]: I think these classify as unrelated changes. Consider a separate commit.

**Comment by azagrebin** [comment]
Thanks for the review @GJL 
I addressed the comment, any other comments?

**Comment by azagrebin** [comment]
merged into master by 9f6b2044786d09b3b0d2484a2c62cc7231413f22
